internal/agentsession/codex_client.go

```
@@ -0,0 +1,201 @@
+	stopOnce  sync.Once
+	requestMu sync.Mutex
+	nextID    int
+	emit      func(Event)
+}
+
+func startCodexClient(ctx context.Context, environment map[string]string, emit func(Event)) (*codexClient, error) {
+	executable := resolveCodexExecutable(environment)
+	if executable == "" {
+		return nil, errCodexUnavailable
+	}
+	command := exec.Command(executable, "app-server", "--stdio")
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)